core/drivers/crypto/se050/adaptors/utils/scp_config.c

```diff
@@ -0,0 +1,396 @@
+	se050_host_key_store_get_key(&ctx->host_ks, dek_object,
+				     dek, &dek_len, &dek_bit_len);
+
+	st = sss_se05x_key_store_set_key(&ctx->ks, &ko, dek, AES_KEY_LEN_nBYTE,
+					 AES_KEY_LEN_nBYTE * 8, NULL, 0);
+	if (st != kStatus_SSS_Success)
+		return kStatus_SSS_Fail;
+
+	st = sss_se05x_cipher_one_go(&symm, NULL, 0, key, enc, enc_len);
+	if (st != kStatus_SSS_Success)
+		return kStatus_SSS_Fail;
+
+	if (symm.keyObject)
+		sss_se05x_symmetric_context_free(&symm);
+
+	sss_se05x_key_object_free(&ko);
+
+	/* SE050: BUG: these transient objects must be deleted */
+	Se05x_API_DeleteSecureObject(&ctx->session.s_ctx, id);
+
+	return kStatus_SSS_Success;
+}
+
+static sss_status_t prepare_key_data(uint8_t *key, uint8_t *cmd,
+				     struct sss_se05x_ctx *ctx, uint32_t id)
+{
+	uint8_t kc[AES_KEY_LEN_nBYTE] = { 0 };
+	sss_status_t status = kStatus_SSS_Fail;
+
+	/* GP key type AES */
+	cmd[0] = PUT_KEYS_KEY_TYPE_CODING_AES;
+	/* Length of the 'AES key data' */
+	cmd[1] = AES_KEY_LEN_nBYTE + 1;
+	/* Length of 'AES key' */
+	cmd[2] = AES_KEY_LEN_nBYTE;
+	/* Length of key check  */
+	cmd[3 + AES_KEY_LEN_nBYTE] = CRYPTO_KEY_CHECK_LEN;
+
+	status = encrypt_key_and_get_kcv(&cmd[3], kc, key, ctx, id);
+	if (status != kStatus_SSS_Success)
+		return status;
+
+	memcpy(&cmd[3 + AES_KEY_LEN_nBYTE + 1], kc, CRYPTO_KEY_CHECK_LEN);
+
+	return kStatus_SSS_Success;
+}
+
+sss_status_t se050_scp03_prepare_rotate_cmd(struct sss_se05x_ctx *ctx,
+					    struct s050_scp_rotate_cmd *cmd,
+					    struct se050_scp_key *keys)
+
+{
+	sss_status_t status = kStatus_SSS_Fail;
+	size_t kcv_len = 0;
+	size_t cmd_len = 0;
+	uint8_t key_version = 0;
+	/* order of elements in the array matters */
+	uint8_t *key[] = { [0] = keys->enc,
+			   [1] = keys->mac,
+			   [2] = keys->dek,
+	};
+	uint32_t oid = 0;
+	size_t i = 0;
+
+	/* add version to replace in the header */
+	key_version = ctx->open_ctx.auth.ctx.scp03.pStatic_ctx->keyVerNo;
+
+	/* packet for SCP03 keys provision: key_version to replace */
+	cmd->cmd[cmd_len] = key_version;
+	cmd_len += 1;
+
+	cmd->kcv[kcv_len] = key_version;
+	kcv_len += 1;
+
+	for (i = 0; i < ARRAY_SIZE(key); i++) {
+		status = se050_get_oid(kKeyObject_Mode_Transient, &oid);
+		if (status != kStatus_SSS_Success)
+			goto error;
+
+		status = prepare_key_data(key[i], &cmd->cmd[cmd_len], ctx, oid);
+		if (status != kStatus_SSS_Success)
+			goto error;
+
+		memcpy(&cmd->kcv[kcv_len],
+		       &cmd->cmd[cmd_len + 3 + AES_KEY_LEN_nBYTE + 1],
+		       CRYPTO_KEY_CHECK_LEN);
+
+		cmd_len += (3 + AES_KEY_LEN_nBYTE + 1 + CRYPTO_KEY_CHECK_LEN);
+		kcv_len += CRYPTO_KEY_CHECK_LEN;
+	}
+
+	cmd->cmd_len = cmd_len;
+	cmd->kcv_len = kcv_len;
+
+	return kStatus_SSS_Success;
+error:
+	EMSG("error preparing scp03 rotation command");
+
+	return kStatus_SSS_Fail;
+}
+
+/*
+ * @param keys
+ *
+ * @return sss_status_t
+ */
+static sss_status_t get_ofid_key(struct se050_scp_key *keys)
+{
+#ifdef CFG_CORE_SE05X_OEFID
```

Comment:
```c
	if (!IS_ENABLED(CFG_CORE_SE05X_OEFID))
		return kStatus_SSS_Fail;

	...

Reply:
CFG_CORE_SE05X_OEFID is an hex value; is  the above still applicable?

Reply:
From this P-R, CFG_CORE_SE05X_OEFID is a binary y/n value:
In **core/crypto.mk**:
```
CFG_CORE_SE05X_OEFID ?= n
```

But yes, looking at the line below, it seems the default value set in **core/crypto** is not relevant.

Anyway, prefer avoid `#if`-like statement where possible, so here, prefer:
```
	if (!CFG_CORE_SE05X_OEFID)
		return kStatus_SSS_Fail;

Reply:
better. thanks!

Reply:
If `CFG_CORE_SE05X_OEFID` is an hex value, we can't have `CFG_CORE_SE05X_OEFID ?= n` in `core/crypto.mk`.

Reply:
yes, changed it to 0